data_safe_haven/infrastructure/project_manager.py

```
@@ -365,23 +375,33 @@ def update_dsh_pulumi_project(self) -> None:
         all_config_dict = {
             key: item.value for key, item in self.stack_all_config.items()
         }
-        self.pulumi_project.encrypted_key = self.stack.workspace.stack_settings(
-            stack_name=self.stack_name
-        ).encrypted_key
         self.pulumi_project.stack_config = all_config_dict
 
+    def update_dsh_pulumi_config(self) -> None:
+        """Update persistent data in the DSHPulumiProject object"""
+        if self.pulumi_config.encrypted_key is None:
+            self.pulumi_config.encrypted_key = self.stack.workspace.stack_settings(
+                stack_name=self.stack_name
+            ).encrypted_key
+
```

Should this check if the stack encrypted key is different from the one in the Pulumi config? We expect them to be the same, but should we raise an Exception if they're not?

---

Could do, that should never happen though.

---

"Should never happen" feels like a great place to raise an exception tbh :)

---

Fixed in b8ac270c